control-plane/p4RuntimeSerializer.cpp

```diff
@@ -2174,5 +1393,44 @@ void serializeP4RuntimeIfRequired(const IR::P4Program* program,
     }
 }
 
+P4RuntimeSerializer::P4RuntimeSerializer() {
+    registerArch("v1model", new ControlPlaneAPI::Standard::V1ModelArchHandlerBuilder());
+    // TODO(antonin): implement P4RuntimeArchHandlerIface for PSA and register
+    // it here.
+}
+
+P4RuntimeSerializer*
+P4RuntimeSerializer::get() {
+    static P4RuntimeSerializer instance;
+    return &instance;
+}
+
+cstring
+P4RuntimeSerializer::resolveArch(const CompilerOptions& options) {
+    if (auto arch = getenv("P4C_DEFAULT_ARCH")) {
+        return cstring(arch);
+    } else if (options.arch != "") {
+        return options.arch;
+    } else {
+        return "v1model";
+    }
```

Comment:
Isn't this already resolved by the backend you use?

Reply:
A backend can support multiple architectures. For example ideally the bmv2 backend would be generic and support all architectures.

Reply:
Unfortunately this code runs before the IR reaches a back-end.

Reply:
currently each bmv2 backend is a separate binary.